dms/controllers/portal.py

```diff
@@ -124,11 +116,16 @@ def portal_my_dms_directory(
                 search_domain = OR([search_domain, [("name", "ilike", search)]])
             domain += search_domain
         # content according to pager and archive selected
-        dms_directory_items = (
-            request.env["dms.directory"]
-            .sudo(request.env.user.id)
-            .search(domain, order=sort_br)
-        )
+        if access_token:
+            dms_directory_items = (
+                request.env["dms.directory"].sudo().search(domain, order=sort_br)
+            )
+        else:
+            dms_directory_items = (
+                request.env["dms.directory"]
+                .sudo(request.env.user.id)
```

Comment:
This seems redundant, isn't it?

```suggestion
```

Reply:
It's very important and related to https://github.com/OCA/dms/pull/85#discussion_r640515228 (with `access_token` is necessary to use `.sudo()` but if not use access_token is very important to apply `request.env.user.id`).

Reply:
Yes, I mean that it's already applied. `request.env["dms.directory"]` already has `request.env.user.id`, so this line is redundant.

Reply:
Ok thanks, changes done.